tests/Integration/ProductHub.IntegrationTests/ProductControllerIntegrationTests.cs

```diff
@@ -1,10 +1,123 @@
+using AutoFixture;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Newtonsoft.Json;
+using ProductHub.Domain.Product;
+using System.Net;
+using System.Net.Http.Json;
+using System.Text;
+
 namespace ProductHub.IntegrationTests;
 
 public class ProductControllerIntegrationTests
+    : IClassFixture<WebApplicationFactory<Program>>
 {
+    private readonly Fixture _fixture;
+    private readonly HttpClient _httpClient;
+    private readonly WebApplicationFactory<Program> _appFactory;
+
+    public ProductControllerIntegrationTests()
+    {
+        _appFactory = new WebApplicationFactory<Program>();
+        _httpClient = _appFactory.CreateClient();
+        _fixture = new Fixture();
+    }
+
+    [Fact]
+    public async Task GetProductsAsync_Success_EndpointReturnsOkResult()
+    {
+        // Arrange
+        var expectedProducts = _fixture.CreateMany<Product>();
+
+        // Act
+        var response = await _httpClient.GetAsync("https://localhost:44363/api/products");
+
+        // Assert
+        var products = await response.Content.ReadFromJsonAsync<IEnumerable<Product>>();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        products.Should().BeEquivalentTo(expectedProducts);
+    }
+
+    [Fact]
+    public async Task GetProductAsync_ValidId_EndpointReturnsOkResult()
+    {
+        // Arrange
+        var productId = _fixture.Create<string>();
```

Comment:
feels more like unit test and less like integration tests, how do we know if the product Id exists